src/reaktoro_pse/core/tests/test_reaktoro_inputs.py

```diff
@@ -118,6 +119,51 @@ def test_with_rkt_sum(build_rkt_state_with_species):
     assert rkt_input.neutrality_ion == "Cl"
 
 
+def test_with_rkt_sum_and_pE(build_rkt_state_with_species_and_pE):
+    """testing if we can construct all constraints correctly if summing with
+    rkt, so inputs should be apprant species, with single empty constraint for CO3-2"""
+    m, rkt_state = build_rkt_state_with_species_and_pE
+    rkt_state.build_state()
+    rkt_state.equilibrate_state()
+    rkt_input = RktInputspec.ReaktoroInputSpec(rkt_state)
+    rkt_input.configure_specs(dissolve_species_in_rkt=True)
+    rkt_input.build_input_specs()
+    input_names = rkt_input.equilibrium_specs.namesControlVariables()
+    print(input_names)
+    input_constraints = rkt_input.equilibrium_specs.namesConstraints()
+    print(input_constraints)
+    expected_names = [
+        "[Cl]",
+        "[C]",
+        "[Na]",
+        "[Mg]",
+        "[Ca]",
+        "[H2O]",
+        "[O]",
+        "[H]",
+        "[H+]",
+        "[e-]",
+    ]
+
+    expected_constraints = [
+        "charge",
+        "C_constraint",
+        "Na_constraint",
+        "Mg_constraint",
+        "Ca_constraint",
+        "H2O_constraint",
+        "O_dummy_constraint",
+        "H_dummy_constraint",
+        "pH",
+        "pE",
+    ]
+
+    for en in expected_names:
+        assert en in input_names
+    for ec in expected_constraints:
```

Comment:
These only check that what you expect to be there is there, i.e. ``expected_names`` is a subset of ``input_names``. You should also check that ``input names`` does not contain things that should not be there.

Adding a check that the lists are of equal length should be sufficient.